toml/src/main/java/com/fasterxml/jackson/dataformat/toml/TomlGenerator.java

```diff
@@ -570,7 +573,12 @@ public void writeNumber(String encodedValue) throws IOException {
 
     @Override
     public void writeNull() throws IOException {
-        throw new UnsupportedOperationException("Nulls are not supported by TOML");
+        if (TomlWriteFeature.FAIL_ON_NULL_WRITE.enabledIn(_tomlFeatures)) {
+            throw new TomlStreamWriteException("TOML null writing disabled (FAIL_ON_NULL_WRITE)", this);
```

Comment:
Maybe add `TomlWriteFeature.` prefix in error message to give context (bit easier for users to find)?